eng/targets/CSharp.Common.targets

```diff
@@ -36,7 +36,7 @@
 
     <!-- Enable .NET code style analysis during build for src projects. -->
     <!-- Workaround bug where turning this on produces warnings in VS -->
```

Comment:
@pranavkm do we have an issue tracking turning this on unconditionally❔

Reply:
https://github.com/dotnet/roslyn/issues/54867

Reply:
nit: might be worth adding that to this comment

now hidden:
> And, was I correct we should enforce code style when building w/ [newer] desktop `msbuild`❔

Reply:
Nope -- looks like the BuildInVS check is sufficient. Added hte code comment.